src/sequential.cpp

```diff
@@ -326,6 +326,24 @@ void Sequential::backward()
                               this->input_state_update);
 }
 
+void Sequential::smoother()
+/*
+ */
+{
+    // Hidden layers
+    for (auto layer = this->layers.begin(); layer != this->layers.end();
+         layer++) {
+        auto *current_layer = layer->get();
+
+        if (auto *slstm_layer = dynamic_cast<SLSTM *>(current_layer)) {
```

Comment:
This is too complicated. Here is something to consider
```c++
current_layer->get_layer_type() == LayerType::SLSTM
```
Same for the SLinear

Reply:
Doing this, I get the error: 
```
no member named 'smoother' in 'BaseLayer'
                  current_layer->smoother(*this->temp_states);
```
This is because I don't have smoother function in the `BaseLayer`

Reply:
The idea is that we need to make sure the current layer is SLSTM in order to perform the `dynamic_cast`. Othewise , it will return a nullptr.
```c++
if (current_layer->get_layer_type() == LayerType::SLSTM) {
     auto *slstm_layer = dynamic_cast<SLSTM *>(current_layer)
     ...
}
```

Reply:
I will follow your suggestion